project-koku/koku
# [COST-3827] Make masu sources PATCH api for external id

## Jira Ticket

[COST-3827](https://issues.redhat.com/browse/COST-3827)

## Description

This change will provide a limited PATCH API for the masu admin sources API. It will allow updating credentials for AWS sources, specifically to add an external id.

Also added is org ID representation and filtering for the masu sources admin api. 

## Testing

1. Checkout Branch
2. Create an AWS source
3. Navigate to http://127.0.0.1:5042/api/cost-management/v1/sources/{uuid}/ 
    1. Make a PATCH HTTP call to this endpoint with the following 
    
```    
{
    "authentication": {
        "credentials": {
            "role_arn": "arn:aws:iam::123456789101:role/CostWithExternalID",
            "external_id": "1234567890"
        }
    }
}
```


4. Observe the update on the source and provider record

## Notes

...
